File: app/src/main/java/za/co/woolworths/financial/services/android/ui/fragments/faq/web/WebFragment.java
package za.co.woolworths.financial.services.android.ui.fragments.faq.web;


import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.lifecycle.ViewModelProviders;

import android.view.View;

import com.awfs.coordination.BR;
import com.awfs.coordination.R;
import com.awfs.coordination.databinding.ActivityFaqdetailsWebBinding;

import za.co.woolworths.financial.services.android.ui.base.BaseFragment;
import za.co.woolworths.financial.services.android.ui.fragments.faq.FAQViewModel;

public class WebFragment extends BaseFragment<ActivityFaqdetailsWebBinding, FAQViewModel> {

	private String mUrl;

	FAQViewModel faqViewModel;

	@Override
	public FAQViewModel getViewModel() {
		return faqViewModel;
	}

	@Override
	public int getBindingVariable() {
		return BR.viewModel;
	}

	@Override
	public int getLayoutId() {
		return R.layout.activity_faqdetails_web;
	}

	@Override
	public void onCreate(@Nullable Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		Bundle bundle = getArguments();
		if (bundle != null) {
			mUrl = bundle.getString("web_url");
		}
		faqViewModel = ViewModelProviders.of(this).get(FAQViewModel.class);
	}

	@Override
	public void onViewCreated(View view, Bundle savedInstanceState) {
		super.onViewCreated(view, savedInstanceState);
		setToolbarBackgroundDrawable(R.drawable.appbar_background);
		showBackNavigationIcon(true);
		setTitle(getString(R.string.drawer_faq));
		showToolbar();
		bindDateWithUI();
	}

	public void bindDateWithUI() {
		getViewDataBinding().faqWeb.getSettings().setJavaScriptEnabled(true);
		getViewDataBinding().faqWeb.loadUrl(mUrl);
	}
}

File: app/src/main/java/za/co/woolworths/financial/services/android/ui/fragments/faq/web/WebFragment.kt
package za.co.woolworths.financial.services.android.ui.fragments.faq.web

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.awfs.coordination.R
import kotlinx.android.synthetic.main.activity_faqdetails_web.*
import za.co.woolworths.financial.services.android.ui.activities.account.MyAccountActivity
import za.co.woolworths.financial.services.android.ui.activities.dashboard.BottomNavigationActivity
import za.co.woolworths.financial.services.android.ui.activities.dashboard.BottomNavigator

class WebFragment : Fragment() {
    private var mUrl: String? = null
    private var mBottomNavigator: BottomNavigator? = null

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (context is BottomNavigationActivity)
            mBottomNavigator = context
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        arguments?.apply {
            mUrl = getString("web_url", "")
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.activity_faqdetails_web, container, false)
    }


    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        setupToolbar()

        bindDateWithUI()
    }

    private fun setupToolbar() {
        (activity as? BottomNavigationActivity)?.apply {
            setToolbarBackgroundDrawable(R.drawable.appbar_background)
            showBackNavigationIcon(true)
            setTitle(getString(R.string.drawer_faq))
            showToolbar()
        }
        if (activity is MyAccountActivity)
            (activity as? MyAccountActivity)?.setToolbarTitle(activity?.resources?.getString(R.string.woolworths_online))
    }

    private fun bindDateWithUI() {
        faqWeb?.settings?.javaScriptEnabled = true
        faqWeb?.loadUrl(mUrl)
    }
}